pkg/kubelet/kubelet_node_status.go

```diff
@@ -370,13 +372,21 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	// apiserver cache (the data might be slightly delayed but it doesn't
 	// seem to cause more conflict - the delays are pretty small).
 	// If it result in a conflict, all retries are served directly from etcd.
-	opts := metav1.GetOptions{}
+	var node *v1.Node
+	var err error
 	if tryNumber == 0 {
-		util.FromApiserverCache(&opts)
-	}
-	node, err := kl.heartbeatClient.Nodes().Get(string(kl.nodeName), opts)
-	if err != nil {
-		return fmt.Errorf("error getting node %q: %v", kl.nodeName, err)
+		node, err = kl.nodeInfo.GetNodeInfo(string(kl.nodeName))
+		if err != nil {
+			return fmt.Errorf("error getting node from cache %q: %v", kl.nodeName, err)
+		}
+		if len(node.Status.Conditions) > 0 && node.Status.Conditions[0].LastHeartbeatTime.Before(&lastHeartbeatTime) {
```

Comment:
don't assume the first condition in the list is one set by the node with a heartbeat time... look for a specific condition we know the kublet sets

Reply:
@wojtek-t how does this interact with your proposal to move heartbeat timestamps out of the node status?

Reply:
ok